app/controllers/application_controller.rb

```diff
@@ -71,14 +73,19 @@ def check_banned
 
   def set_chat
     if current_user && current_user.system
-      current_user.system.wormhole? ? @system_users = [] : @system_users = User.where("online > 0").where(system: current_user.system)
-      @global_messages = ChatMessage.includes(:user).where(chat_room: ChatRoom.where(chatroom_type: :global).first).last(20)
+      @system_users = []
+      @system_users = current_user.system.users.is_online unless current_user.system.wormhole?
+      @global_messages = ChatMessage.includes(:user).where(chat_room: ChatRoom.global.first).last(20)
```

Comment:
Can we do `ChatRoom.global.first.chat_messages.includes(:user).last(20)` here instead?

Reply:
yeah... that's much better - this is the result of search/replace